### PR TITLE
Update dev docs for go mod changes

### PR DESCRIPTION
**Pull request by bentito**
Remove refs to `dep` and replace with instruction for `go mod` and general installation via `git clone`.

### CODE REVIEW

File: Documentation/dev/developer-guide.md [hunk 1]
```diff
@@ -111,30 +102,33 @@ make integration REPORTING_OPERATOR_IMAGE_TAG=pr-1234 METERING_OPERATOR_IMAGE_TA
...
 root of the repository, and to ensure everyone is using the same dependencies,
 and ensure that if dependencies disappear, we commit the contents of `vendor/`
 into git.
 
 ### Adding new dependencies
 
-To add a new dependencies, you can generally follow the dep documentation.
-Start by reading [https://golang.github.io/dep/docs/daily-dep.html](https://golang.github.io/dep/docs/daily-dep.html)
-and you it should cover the most common things you'll be using dep for.
+To add a new dependencies, you can generally do the following:
```
Comment (timflannagan1) [inline Documentation/dev/developer-guide.md:115]:
```suggestion
To add new dependencies, you can generally do the following:
```

File: Documentation/dev/developer-guide.md [hunk 2]
```diff
@@ -147,4 +141,8 @@ helm template CHART_DIR -x PATH_TO_TEMPLATE/file.yaml
 ## Developer install
 
 Developers should generally use the [manual-install guide](../manual-install.md) as it offers the most flexibility when installing.
-If you need a minimal storage configuration with no external dependencies, use the [manifests/metering-config/hdfs-minimal.yaml](../../manifests/metering-config/hdfs-minimal.yaml) example configuration.
+If you need a minimal storage configuration with no external dependencies, use the [manifests/metering-config/hdfs-minimal.yaml](manifests/metering-config/hdfs-minimal.yaml) example configuration.
```
Comment (timflannagan1) [inline Documentation/dev/developer-guide.md:144]: That path is local to the repository so changing that path leads to a `Page not found` error from Github.

File: Documentation/dev/developer-guide.md [hunk 3]
```diff
@@ -111,30 +102,33 @@ make integration REPORTING_OPERATOR_IMAGE_TAG=pr-1234 METERING_OPERATOR_IMAGE_TA
...
 and ensure that if dependencies disappear, we commit the contents of `vendor/`
 into git.
 
 ### Adding new dependencies
 
-To add a new dependencies, you can generally follow the dep documentation.
-Start by reading [https://golang.github.io/dep/docs/daily-dep.html](https://golang.github.io/dep/docs/daily-dep.html)
-and you it should cover the most common things you'll be using dep for.
+To add a new dependencies, you can generally do the following:
+```
```
Comment (timflannagan1) [inline Documentation/dev/developer-guide.md:116]:
```suggestion
```bash
```

File: Documentation/dev/developer-guide.md [hunk 4]
```diff
@@ -29,19 +29,10 @@ $ pip3 install operator-courier
...
-The repository path must be located at:
-
-```
-$GOPATH/src/github.com/operator-framework/operator-metering
-```
-
-When cloning this repository, you can run the following commands to ensure that the project files are stored in the right location:
+Clone this repository:
 
 ```
```
Comment (timflannagan1) [inline Documentation/dev/developer-guide.md:34]:
```suggestion
```bash
```

File: Documentation/dev/developer-guide.md (hunk 3 above)
Comment (timflannagan1) [reply Documentation/dev/developer-guide.md:116]: It looks like we're pretty inconsistent with adding syntax highlighting to code blocks, so we can either update all of the tag all of the codeblocks in this doc, or just ignore this suggestion.

File: Documentation/dev/developer-guide.md [hunk 5]
```diff
@@ -111,30 +102,33 @@ make integration REPORTING_OPERATOR_IMAGE_TAG=pr-1234 METERING_OPERATOR_IMAGE_TA
 
 ## Go Dependencies
 
-We use [dep](https://golang.github.io/dep/docs/introduction.html) for managing
+We use `go mod` for managing
```
[the remark this answers is not in the document]
Comment (timflannagan1) [reply Documentation/dev/developer-guide.md:105]: Do we need to provide an example `go env` configuration, i.e. ensure `GO111MODULE` is not set to "off"?

File: Documentation/dev/developer-guide.md (hunk 2 above)
Comment (bentito) [reply Documentation/dev/developer-guide.md:144]: Actually I pulled this change in accidentally, didn't make it explicitly. I'll change it back

Comment (bentito) [reply Documentation/dev/developer-guide.md:144]: I was actually using Github's editor, it must have tweaked the URL

File: Documentation/dev/developer-guide.md (hunk 3 above)
Comment (bentito) [reply Documentation/dev/developer-guide.md:116]: updated all in this doc

File: Documentation/dev/developer-guide.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (bentito) [reply Documentation/dev/developer-guide.md]: I think the whole go env might be hard to prove out.  Not having GO111MODULE set at all will work. Seems odd to start mentioning things people might possibly set unless we need something that is not the default setting for a go installation.

File: Documentation/dev/developer-guide.md [hunk 6]
```diff
@@ -104,47 +95,53 @@ The steps can be broken down into:
...
 ### Adding new dependencies
 
-To add a new dependencies, you can generally follow the dep documentation.
-Start by reading [https://golang.github.io/dep/docs/daily-dep.html](https://golang.github.io/dep/docs/daily-dep.html)
-and you it should cover the most common things you'll be using dep for.
+To add a new dependency, you can do the following:
+```bash
+go get <dependency_repo_url>@<version_of_dependency>
+```
+You can learn more about version specification here: (using-go-modules)[https://blog.golang.org/using-go-modules].
```
[the remark this answers is not in the document]
Comment (bentito) [reply Documentation/dev/developer-guide.md]: fixed

**Comment by bentito** [comment]
Added link text and change dependency list to mention Go 1.13 or higher.

**Pull request merged**